Automattic/jetpack
# Jetpack Backup: update logos

This PR updates the logos and icons in the Backup plugin to match the new _Jetpack VaultPress Backup_ branding.

Fixes 1202858161751496-as-1203427126857118

#### Changes proposed in this Pull Request:
- Added the `JetpackVaultPressBackupLogo` component to the `jetpack-components` package.
- Added the possibility to pass a React element to the `icon` prop of the `PricingCard` component, in the `jetpack-components` package.
- Added the possibility to pass a logo (as a React element) to the connection screen components, in the `jetpack-connection` package. 
- Updated the logo and icon in the Backup upsell page.
- Updated the top logo in the backup dashboard.

#### Other information:


#### Jetpack product discussion
Context: p1HpG7-jkl-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:

- Spin up a test site
- Build and install the Jetpack VaultPress Backup plugin
- Visit the `Jetpack > VaultPress Backup` section
- Note the logos and icons are updated (see captures below)
- Purchase a subscription to VaultPress Backup
- Notice that the top logo in the backup dashboard is updated (see captures below)

#### Captures

_Upsell page_
| Before | After |
|--------|------|
| <img width="500" alt="Screen Shot 2022-12-06 at 3 26 40 PM" src="https://user-images.githubusercontent.com/1620183/206224241-b070e789-7b98-4738-9353-ccaa5b28ee20.png"> | <img width="500" alt="Screen Shot 2022-12-06 at 4 33 54 PM" src="https://user-images.githubusercontent.com/1620183/206224245-3f463ac0-2750-489b-84d0-c764dbec9efb.png"> |


_Backup dashboard_
| Before | After |
|--------|------|
| <img width="1365" alt="Screen Shot 2022-12-07 at 10 25 07 AM" src="https://user-images.githubusercontent.com/1620183/206224666-75ffcbb2-5d06-42f8-850a-b60e0841a65c.png"> | <img width="1363" alt="Screen Shot 2022-12-07 at 10 28 51 AM" src="https://user-images.githubusercontent.com/1620183/206224668-57871f9b-4952-42c2-b78b-be75d6773848.png"> |
